osmo_jupyter/plot.py

```diff
@@ -0,0 +1,224 @@
+    >>> get_rgb_columns(df, '*_mean', ['Timestamp'])
+    <DataFrame with columns ['r', 'g', 'b', 'Timestamp'] >
+
+    Params:
+        df: pandas.DataFrame of source data
+        pattern: string with an asterisk ("*") in place of the color letter, eg. '*_mean'
+        other_columns_to_include: list of column names to be included
+
+    '''
+    other_columns_to_include = other_columns_to_include or []
+
+    color_columns = [
+        pattern.replace('*', color)
+        for color in COLOR_CHANNELS
+    ]
+
+    all_desired_columns = color_columns + other_columns_to_include
+    rename_map = {
+        original_column: new_column
+        for original_column, new_column in zip(color_columns, COLOR_CHANNELS)
+    }
+    return df[all_desired_columns].rename(columns=rename_map)
+
+
+def get_layout_with_annotations(
+    x_axis_title='Time',
+    y_axis_title=None,
+    colors_on_separate_axes=False,
+    events=None,
+    **additional_layout_params
+):
+    ''' Get a go.Layout instance prepared for our standard graphing needs
+
+    Notes re. our standard layout:
+    * The primary Y axis (y0) is reserved for annotations. This is done to guarantee that annotations show
+    * If colors are on separate axes, they will be on the next consecutive axes, otherwise they will be on yaxis 2
+    * You can use additional axes for plotting other things by passing in additional params
+
+    Params:
+        x_axis_title: title to use for X axis, default 'Time'
+        y_axis_title: title to use for Y axis where colors are plotted. Required if colors_on_separate_axes is False.
+            If colors_on_separate_axes is True, it will be used as a suffix for the y axis titles "r", "g", and "b"
+        colors_on_separate_axes: whether colors (r,g and b) should be displayed on separate Y axes
+        events: dictionary of {x_axis_value: "annotation"} which will be used to annotate the chart
+        additional_layout_params: These will be passed to go.Layout. These can, for instance, be the result of
+            axis_kwargs() to add support for graphing more items.
+    '''
+    events = events or {}
+    if not colors_on_separate_axes and not y_axis_title:
+        raise ValueError('If colors_on_separate_axes is False you must provide a y_axis_title')
+
+    annotation_y_axis = {
+        'yaxis': {
+            'title': '',
+            'range': [0, 1],
+            'visible': False,
+        }
+    }
+
+    def get_color_y_axis_title(color):
+        if not colors_on_separate_axes:
+            return y_axis_title
+
+        # "blue" axis uses a line break to not overlap unnecessarily with green
+        line_break = '<br>' if color == 'b' else ''
+
+        # If colors are on separate axes and a Y axis title is provided, use it as a suffix
+        suffix = f' {y_axis_title}' if y_axis_title else ''
+        return f'{line_break}{color}{suffix}'
+
+    color_y_axes = _add_dicts(
```

Comment:
this block is a bit hard to parse ... could help to break it out into separate variables, e.g.

```
colors_on_same_axis_kwargs = ...
colors_on_separate_axes_kwargs = ...

color_y_axes = colors_on_separate_axes_kwargs if colors_on_separate_axes else colors_on_same_axis_kwargs
```

Reply:
doing that. will be in next push